### PR TITLE
Fix configuration, add validations for project filtering

### PR DESCRIPTION
**Pull request by lukaszgryglicki**
This will require some deployment changes, 2 scenarios:
- You want all projects, you need to specify `GERRIT_PROJECT=''` and `GERRIT_PROJECT_FILTER=0`.
- You want a specific project, you need to specify `GERRIT_PROJECT='proj-name'` and `GERRIT_PROJECT_FILTER=1`. 

More details https://jira.linuxfoundation.org/browse/DA-4801?focusedCommentId=719838&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-719838

cc @fayazg @1010sachin @priyankakale15 @khalifapro (please ping whoever is working on actual Gerrit deployment details)

Signed-off-by: Łukasz Gryglicki <lgryglicki@cncf.io>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -16,4 +17,4 @@ ENV ELASTIC_LOG_PASSWORD='<ELASTIC-LOG-PASSWORD>'
 RUN apk update && apk add --no-cache bash
 RUN ls -ltra
 COPY gerrit ./
-CMD ./gerrit --gerrit-disable-host-key-check=true --gerrit-url=${GERRIT_URL} --gerrit-project=${GERRIT_PROJECT} --gerrit-es-url=${ES_URL} --gerrit-user=${GERRIT_USERNAME} --gerrit-ssh-key=${GERRIT_SSH_KEY}
+CMD ./gerrit --gerrit-disable-host-key-check=true --gerrit-url=${GERRIT_URL} --gerrit-project=${GERRIT_PROJECT} --gerrit-project-filter=${GERRIT_PROJECT_FILTER} --gerrit-es-url=${ES_URL} --gerrit-user=${GERRIT_USERNAME} --gerrit-ssh-key=${GERRIT_SSH_KEY}
```
Comment (ajinkyan83) [inline Dockerfile:20]: `--gerrit-project-filter` - it does not need to have a value, I mean it is not mandatory. correct?
If this is the case, then the below env variables need not be set, I think.

File: cmd/lambda/lambda.go [hunk 2]
```diff
@@ -41,6 +41,10 @@ func HandleRequest(ctx context.Context) {
               Name: aws.String("GERRIT_PROJECT"),
               Value: aws.String("GERRIT_PROJECT"),
             },
+            &ecs.KeyValuePair{
+              Name: aws.String("GERRIT_PROJECT_FILTER"),
+              Value: aws.String("GERRIT_PROJECT_FILTER"),
+            },
```
Comment (ajinkyan83) [inline cmd/lambda/lambda.go:47]: This need not be set.

Comment (lukaszgryglicki) [reply cmd/lambda/lambda.go:47]: Explained in the previous comment - I prefer to be more explicit, so we know what is needed, you can just put `0` or `false` which does the same as not sending at all.

**Review by ajinkyan83 — APPROVED** [review]
/lgtm

**Pull request merged**